compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -1974,16 +1974,16 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                         (ty::Tuple(_), ty::Tuple(_)) => {}
                         (ty::Tuple(_), _) if expected.tuple_fields().count() == 1 => {
                             if let Ok(code) = self.tcx.sess().source_map().span_to_snippet(span) {
-                                if let Some(code) =
-                                    code.strip_prefix('(').and_then(|s| s.strip_suffix(')'))
-                                {
-                                    err.span_suggestion(
-                                        span,
-                                        "use a trailing comma to create a tuple with one element",
-                                        format!("({},)", code),
-                                        Applicability::MaybeIncorrect,
-                                    );
-                                }
+                                let code_stripped = code
+                                    .strip_prefix('(')
+                                    .and_then(|s| s.strip_suffix(')'))
+                                    .unwrap_or(&code);
```

Comment:
Can you explain why you changed this? It's not apparent to me what the change in behavior is.

Reply:
Certainly, so with a bit of context: the `if` statement which guarded this previously meant we'd suggest `(xyz)` --> `(xyz,)`, if a 1-tuple was expected (and the parentheses were already there).

Removing this guard means we always suggest the parentheses + comma, when a 1-tuple is expected, even if the parentheses weren't already there, i.e. `xyz -> (xyz,)`. It's a slightly larger brush stroke than the rest of the PR, which aims just to suggest tuples in argument positions (whereas this applies anywhere a 1-tuple is expected).

Reply:
I see; thanks for the explanation. Is there a reason this needs to be bundled with the other changes in this PR? This code was originally added in #86116, which says this:

> i.e. I have added a suggestion to add a trailing comma to create a 1-tuple. This suggestion is only issued if a 1-tuple is expected and the expression ((1) in the example above) is surrounded by parentheses and does not already have a tuple type. In this situation, I'd say that it is pretty likely that the user meant to create a tuple.

I don't think there's any downside to expanding the suggestion to apply no matter whether there are parentheses (especially because the check is fragile to formatting); but could you create a separate PR (you can assign me on it) to keep the changes separate?

Reply:
My main reason for bundling it in was to also cover 1-tuples, but happy to split it apart seing as it's a slightly different / existing case already - moved to #91530